endogenous/scripts/end_transport.py

```diff
@@ -111,36 +113,92 @@ def transport(string):
     #add solar PV generator with data from one country
     df_solar = pd.read_csv(string + 'resources/pv_optimal.csv', sep=';', index_col=0)
     df_solar.index = pd.to_datetime(df_solar.index)
-    CF_solar = df_solar['ESP'][[hour.strftime("%Y-%m-%dT%H:%M:%SZ") for hour in network.snapshots]]
+    CF_solar = df_solar[country][[hour.strftime("%Y-%m-%dT%H:%M:%SZ") for hour in network.snapshots]]
     solar = df_tech.loc['solar']
     solar = df_tech.loc['solar'].set_index(['parameter'])
-    capital_cost_solar = annuity(solar.at['lifetime','value'], 0.07)*solar.at['investment','value']*1000*+solar.at['FOM','value']/100 # in €/MW
+    capital_cost_solar = solar.at['investment','value']*1000*(annuity(solar.at['lifetime','value'], 0.07)+solar.at['FOM','value']/100) # in €/MW
     # capital_cost_solar = annuity(25, 0.07)*425000*(1+0.03) # in €/MW
     network.add("Generator",
                 "solar",
                 bus="electricity bus",
                 p_nom_extendable=True,
                 carrier="solar",
-                #p_nom_max=1000, #maximum capacity can be limited due to environmetal constraints
+                #p_nom_max=1000, #maximum capacity can be limited due to environmental constraints
                 capital_cost = capital_cost_solar,
-                marginal_cost = 0.01,
+                marginal_cost = solar.at['VOM','value'],
                 p_max_pu = CF_solar)
     
-    
+    network.add("Bus", "battery bus")
+
+    df_costs = pd.read_csv(string + 'resources/costs_' + str(Nyear) + '.csv', sep=',', index_col=0)
+    #set first column as second index and keep original index
+    df_costs.set_index(df_costs.columns[0],append=True,inplace=True)
+    #set parameter/value names as column name, min_count=1 fills not existing values with NaN
+    costs = df_costs.loc[:,"value"].unstack(level=1).groupby("technology").sum(min_count=1)
+    network.add("Store", "battery storage",
+                bus="battery bus",
+                e_nom_extendable=True,
+                e_cyclic = True,
+                capital_cost=costs.at['battery storage', 'investment'],
+                lifetime = costs.at['battery storage', 'lifetime'])
+    network.add("Carrier",'battery')
+    network.add("Link",
```

Comment:
this link is missing a representation of when the battery cars are parked and connected to the grid so that they can be charged, check @ricnogfer implementation when he added it

Reply:
I see this is addressed here: https://github.com/ricnogfer/pypsa-eur-sec_land_transport/pull/24